pkg/crosscluster/producer/replication_manager.go

```diff
@@ -450,26 +450,42 @@ func (r *replicationStreamManagerImpl) AuthorizeViaJob(
 	return nil
 }
 
-func (r *replicationStreamManagerImpl) AuthorizeViaReplicationPriv(ctx context.Context) error {
-	err := r.evalCtx.SessionAccessor.CheckPrivilege(ctx,
+// AuthorizeViaReplicationPriv ensures the user has the REPLICATIONSOUCE privilege. If tableNames is passed, then table level auth is tried.
+func (r *replicationStreamManagerImpl) AuthorizeViaReplicationPriv(
+	ctx context.Context, tableNames ...string,
+) (err error) {
+	defer func() {
```

Comment:
style nit: I find control flow logic in a defer to be confusing. I try to restrict defer logic to cleanup logic. I think the code reads clearer if you use an internal lambda.

```
authorize := func() (error, authorized) {
   ...
   if err == nil {
      return nil, true
   }
}
var err error
err, r.authorized = authorize()
return err
```